app/models/training_period.rb

```diff
@@ -197,4 +198,10 @@ def schedule_applicable_for_trainee
 
     errors.add(:schedule, "Only mentors can be assigned to replacement schedules") if schedule.replacement_schedule?
   end
+
+  def school_consistency
+    return if school_partnership.school == trainee&.school
```

Comment:
Unrelated to this PR but I think we should rename `#trainee` to an unprefixed `#at_school_period` - the word trainee describes a teacher but this is pointing at the time  the teacher was at the school.